tests/integration_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,186 @@
+            result = bq_utils.load_cdm_csv(self.hpo_id,
+                                           cdm_table,
+                                           dataset_id=self.bq_dataset_id)
+            logging.info('Loading table %s.%s' %
+                         (self.bq_dataset_id, hpo_table))
+            job_id = result['jobReference']['jobId']
+            job_ids.append(job_id)
+
+        incomplete_jobs = bq_utils.wait_on_jobs(job_ids)
+        self.assertEqual(len(incomplete_jobs), 0,
+                         'NYC five person load job did not complete')
+        logging.info('All tables loaded successfully')
+
+        # Store query for checking number of rows to delete
+        for ehr in self.deactivated_ehr_participants:
+            pid = ehr[0]
+            for hpo in hpo_table_list:
+                date_column = retraction_info.loc[
+                    retraction_info['table'].str.contains(hpo),
+                    'date_column'].item()
+                query = EXPECTED_ROWS_QUERY.format(
+                    dataset_id=self.bq_dataset_id,
+                    table_id=hpo,
+                    pid_table_id=self.pid_table_id,
+                    pid=pid,
+                    date_column=date_column)
+                row_count_queries.append({
+                    clean_consts.QUERY: query,
+                    clean_consts.DESTINATION_DATASET: self.bq_dataset_id,
+                    clean_consts.DESTINATION_TABLE: hpo
+                })
+
+        # Use query results to count number of expected row deletions
+        expected_row_count = {}
+        for query_dict in row_count_queries:
+            response = client.query(query_dict['query'])
+            result = response.result()
+            expected_row_count[
+                query_dict['destination_table_id']] = result.total_rows
+
+        # Separate check to find number of actual deleted rows
+        q = TABLE_ROWS_QUERY.format(dataset_id=self.bq_dataset_id)
+        q_result = client.query(q)
+        row_count_before_retraction = {}
+        for row in q_result:
+            row_count_before_retraction[row['table_id']] = row['row_count']
+
+        # Perform retraction
+        query_list = retract_deactivated_pids.create_queries(
+            self.project_id, self.ticket_number, self.project_id,
+            self.bq_dataset_id, self.pid_table_id)
+        retract_deactivated_pids.run_queries(query_list, client)
+
+        # Find actual deleted rows
+        q_result = client.query(q)
+        results = q_result.result()
+        row_count_after_retraction = {}
+        for row in results:
+            row_count_after_retraction[row['table_id']] = row['row_count']
+        for table in expected_row_count:
+            self.assertEqual(
+                expected_row_count[table], row_count_before_retraction[table] -
+                row_count_after_retraction[table])
+
+    def tearDown(self):
+        test_util.delete_all_tables(self.bq_dataset_id)
```

Comment:
@amcgrenera-vumc I'm confused.  I changed the tearDown() function to leave the tables so I could see the changes and I don't understand what I am seeing.  This is the change I made.
```

def tearDown(self):
        print(self.bq_dataset_id)
        #test_util.delete_all_tables(self.bq_dataset_id)
        print("be sure to delete these tables later")
```
When I made this change I ran the test locally and this is my deactivated pid table.  Completely reasonable.
```
person_id | deactivated_date   
2         | 2008-02-01
1234567   | 2019-01-01   
1         | 2007-01-01
```

This is my sandboxed table for the `DCXXX_fake_measurement` table.  It only seems to have sandboxed data for person 1 and not any data for person 2.  Based on the `deactivation_date` for person 2, I would expect all of his/her data to be sandboxed as well.
```
measurement_id | person_id | measurement_date | measurement_datetime
 3             | 1         | 2010-11-05       | 2010-11-05 05:00:00 UTC
 2             | 1         | 2008-09-04       | 2008-09-04 05:00:00 UTC
 1             | 1         | 2008-09-04       | 2008-09-04 05:00:00 UTC 
```

This is the person 1 and 2 data in `fake_measurement` at the end of the test.
```
measurement_id    person_id    measurement_date    measurement_datetime
2                    1               2008-09-04    2008-09-04 05:00:00 UTC
1                    1               2008-09-04    2008-09-04 05:00:00 UTC
3                    1               2010-11-05    2010-11-05 05:00:00 UTC
11                   2               2009-06-02    2009-06-02 05:00:00 UTC
5                    2               2009-06-02    2009-06-02 05:00:00 UTC
6                    2               2009-06-02    2009-06-02 05:00:00 UTC
10                   2               2009-06-02    2009-06-02 05:00:00 UTC
7                    2               2009-06-02    2009-06-02 05:00:00 UTC
9                    2               2009-06-02    2009-06-02 05:00:00 UTC
4                    2               2009-06-02    2009-06-02 05:00:00 UTC
8                    2               2009-06-02    2009-06-02 05:00:00 UTC
13                   2               2010-08-03    2010-08-03 05:00:00 UTC
14                   2               2010-08-03    2010-08-03 05:00:00 UTC
12                   2               2010-08-03    2010-08-03 05:00:00 UTC
17                   2               2009-07-16    2009-07-16 05:00:00 UTC
15                   2               2009-08-11    2009-08-11 05:00:00 UTC
16                   2               2009-01-05    2009-01-05 05:00:00 UTC
18                   2               2010-06-18    2010-06-18 05:00:00 UTC
19                   2               2010-06-18    2010-06-18 05:00:00 UTC
20                   2               2008-10-04    2008-10-04 05:00:00 UTC
22                   2               2009-07-31    2009-07-31 05:00:00 UTC
25                   2               2009-07-31    2009-07-31 05:00:00 UTC
28                   2               2009-07-31    2009-07-31 05:00:00 UTC
21                   2               2009-07-31    2009-07-31 05:00:00 UTC
26                   2               2009-07-31    2009-07-31 05:00:00 UTC
27                   2               2009-07-31    2009-07-31 05:00:00 UTC
23                   2               2009-07-31    2009-07-31 05:00:00 UTC
30                   2               2009-07-31    2009-07-31 05:00:00 UTC
31                   2               2009-07-31    2009-07-31 05:00:00 UTC
24                   2               2009-07-31    2009-07-31 05:00:00 UTC
29                   2               2009-07-31    2009-07-31 05:00:00 UTC
32                   2               2009-04-15    2009-04-15 05:00:00 UTC
33                   2               2008-10-09    2008-10-09 05:00:00 UTC
34                   2               2009-11-11    2009-11-11 05:00:00 UTC
35                   2               2010-02-23    2010-02-23 05:00:00 UTC
```
Correct me if I am wrong.  

1. I expected the table `fake_measurement` to be cleaned of all of `person_id in (1, 2)` data based on their deactivation dates.  Meaning, I didn't expect either individual to have data in the table after running the retraction script on the table.  Is this assumption correct?
2. Why did the sandbox table not contain any data for `person_id = 2`?
3.  If the assumption in question 1 is correct, this test needs to be refactored to ensure it is testing what it should be testing.  It is passing, but **if** the output is wrong, it should be failing.
4.  If the output is correct, please point out what I'm not understanding.

Reply:
I'm looking into this. I was adding the additional query like you had suggested above, which took me down this rabbit hole as well. I'll let you know when this is sorted on my end for you to test again.

Reply:
Just to keep you in the loop the row counts weren't accounting for each individual PID, so it was overwriting to zero since PID 12345 didn't exist in any of the tables with the fake data. I'm adjusting the PIDS table to only include PIDs that exist within that table and making sure it isn't overwriting the counts.

Reply:
@amcgrenera-vumc Thank you.  I felt like I was missing something.  So, the pid I needed to be checking for is 1234567?

Reply:
well my test was wrong, in the loop to get the counts:

```        for query_dict in dropped_row_count_queries:
            response = client.query(query_dict['query'])
            result = response.result()
            expected_dropped_row_count[
                query_dict['destination_table_id']] = result.total_rows
```
It would overwrite the counts for the last loop for that table, so it was overwriting to zero since 123456 didn't exist in the data... if that makes sense. I'm refactoring the code now and will let you know when I think I have it working properly.